addons/website/models/website_page.py

```diff
@@ -96,12 +96,19 @@ def _get_most_specific_pages(self):
         ''' Returns the most specific pages in self. '''
         ids = []
         previous_page = None
+        page_keys = self.sudo().search([]).mapped('key')
```

Comment:
I think this part is wrong.

1. The spec of this method is "Returns the most specific pages in self", it's kinda weird to check for pages outside of self (with a `.search([])`)
2. I tested:
  - Create 2 websites, all using the generic `/contactus` page
  - COW that page for website 2
  - Check sitemap of website 1 => `/contactus` does not appear while it should
  (well actually, it works because somehow the sitemap is not recomputed? Another bug? If you delete the sitemap attachment, and regenerate it after, it is buggy)

Actually those steps can also be followed using the page manager instead of the sitemap: the page manager does not show "contact us" for website 1 while it should.

Would not this be fixed by just not doing a `search` but filtering what you have in `self` here? And maybe adapting the docstring to say that this assumes that `self` contains all generic and specific pages with only one `website_id` possible?

Reply:
Good catch, it can't just be about searching for duplicate keys without considering the website for which this method is called.
It should be about adding the `website.website_domain()` in the search, but it's not possible, we don't have any reference of that website in `_get_most_specific_pages`, and we can't retrieve it with `get_current_website` either as one of the caller is calling this on pages found considering `self` as a website..

> Would not this be fixed by just not doing a search but filtering what you have in self here?

That's what I initially do, to later figure that it was not correct. It's exactly why I added the `test_generic_page_diverged_not_shown` test:
- Create 2 websites, all using the generic /contactus page
- COW that page for website 2 and rename its URL to "/whatever"
- On that website 2, autocomplete a URL (new menu, edit mode url..) and type "/cont", it should not list anything, as /contactus (generic) is shadowed by /whatever (speficic).
But if we only consider the pages we have in `self` here, we will only have the generic /contactus and we won't exclude it since /whatever won't be in the recordset to exclude it.

Same probably for the list view in the backend, if you type "/cont" in the search view, you will see the same results.

Generic pages feature... :vomiting_face: 

Anyway, I pushed a diff to add the website in the domain, and I covered this flow within the tests.

Reply:
It's fine as you did but maybe you could still get the website_id by checking the views you have in `self` rather that passing it through the context no? That way the method "would be fixed in stable". No?

Reply:
No because `self` only contains the generic page in the flow we discuss here (that I explained above):

> But if we only consider the pages we have in self here, we will only have the generic /contactus and we won't exclude it since /whatever won't be in the recordset to exclude it.

Reply:
Oh, yeah. I understood that you only had that one + all the other pages but when you pre-filtered base on "/cont", it's dead indeed :grimacing:

Reply:
(I'll squash the commits in a few minutes)

Reply:
Do it now ! :zap:

Reply:
And "voila"